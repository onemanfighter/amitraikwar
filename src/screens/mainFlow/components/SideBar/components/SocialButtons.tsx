import { Box, useColorMode } from '@chakra-ui/react';
import { SOCIAL_MEDIA } from './constants';
import { SocialIcon } from 'react-social-icons';

const SocialButtons = () => {
  const { colorMode } = useColorMode();
  return (
    <Box
      borderRadius={'lg'}
      boxShadow={'lg'}
      margin={'4'}
      padding={'2'}
      gap={3}
      top={-10}
      position={'relative'}
      bg={colorMode === 'light' ? 'green.200' : 'green.700'}
      flexDirection={'row'}
      display={'flex'}
      justifyContent={'center'}
    >
      {SOCIAL_MEDIA.map((social) => (
        <Box key={social.url} paddingX={1}>
          <SocialIcon key={social.url} url={social.url} target="blank" />
        </Box>
      ))}
    </Box>
  );
};

export default SocialButtons;
