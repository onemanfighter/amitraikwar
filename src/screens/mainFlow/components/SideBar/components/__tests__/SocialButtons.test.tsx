import { render } from '@testing-library/react';
import SocialButtons from '../SocialButtons';

jest.mock('react-social-icons', () => ({
  SocialIcon: ({ url }: { url: string }) => <div>{url}</div>,
}));

describe('Social buttons', () => {
  it('should renders correctly', () => {
    const { container } = render(<SocialButtons />);

    expect(container).toMatchSnapshot();
  });
});
