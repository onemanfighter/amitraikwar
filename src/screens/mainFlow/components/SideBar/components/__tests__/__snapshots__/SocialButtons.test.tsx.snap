// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social buttons should renders correctly 1`] = `
<div>
  <div
    class="css-vetxw3"
  >
    <div
      class="css-x988sr"
    >
      <div>
        https://in.linkedin.com/in/amitrai1603
      </div>
    </div>
    <div
      class="css-x988sr"
    >
      <div>
        https://github.com/onemanfighter
      </div>
    </div>
    <div
      class="css-x988sr"
    >
      <div>
        https://github.com/raikwaramit
      </div>
    </div>
    <div
      class="css-x988sr"
    >
      <div>
        https://amitraikwar.medium.com/
      </div>
    </div>
  </div>
</div>
`;
